File: project_euler/005/solution.go
package main 

import "fmt"

func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a % b
	}
	return a
}

func main() {
	r := 1
	for i := 20; i > 1; i -- {
		if r % i != 0 {
			r *= i/gcd(r, i)
		}
	}
	fmt.Println(r)
}

File: project_euler/005/Solution.scala
object Solution{
	def gcd(x : Int, y : Int) : Int = {
		var a = x
		var b = y
		while(b != 0){
			var tmp = b
			b = a % b
			a = tmp
		}
		return a
	}

	def main(args: Array[String]) = {
		var prod = 1
		for(i <- 2 to 19){
			prod *= i/gcd(prod, i)
		}	
		println(prod)
	}
}
